src/prefect/task_runners.py

```diff
@@ -226,14 +227,14 @@ class ConcurrentTaskRunner(BaseTaskRunner):
         >>>     ...
     """
 
-    def __init__(self):
-        # TODO: Consider adding `max_workers` support using anyio capacity limiters
-
+    def __init__(self, max_workers: int =None):
         # Runtime attributes
         self._task_group: anyio.abc.TaskGroup = None
         self._result_events: Dict[UUID, anyio.abc.Event] = {}
         self._results: Dict[UUID, Any] = {}
         self._keys: Set[UUID] = set()
+        self._max_workers = max_workers
+        self._limiter = asyncnullcontext
```

Comment:
```suggestion
        self._limiter = asyncnullcontext()
```

I'm also curious, is there a reason we need to instantiate the limiter in `_start` instead of `__init__`? Does it need the event loop?